src/megameklab/com/printing/PrintTank.java

```diff
@@ -0,0 +1,212 @@
+    protected boolean isCenterlineLocation(int loc) {
+        if (tank.isSuperHeavy()) {
+            return loc == SuperHeavyTank.LOC_FRONT
+                    || loc == SuperHeavyTank.LOC_BODY
+                    || loc == SuperHeavyTank.LOC_REAR;
+        }
+        return loc == Tank.LOC_FRONT
+                || loc == Tank.LOC_BODY
+                || loc == Tank.LOC_REAR;
+    }
+
+    @Override
+    protected String getSVGFileName(int pageNumber) {
+        String subtype;
+        switch (tank.getMovementMode()) {
+            case VTOL:
+                subtype = "vtol";
+                break;
+            case NAVAL:
+            case HYDROFOIL:
+                subtype = "naval";
+                break;
+            case SUBMARINE:
+                subtype = "submarine";
+                break;
+            case WIGE:
+                if (!tank.isSuperHeavy()) {
+                    subtype = "wige";
+                    break;
+                }
+                //fall through
+            default:
+                subtype = "vehicle";
+                break;
+        }
+        String turret;
+        if (tank.hasNoTurret()) {
+            turret = "noturret";
+        } else if (tank.hasNoDualTurret()) {
+            turret = "turret";
+        } else {
+            turret = "dualturret";
+        }
+        // Superheavy VTOLs don't have extra armor sections and use the same template as standard weight
+        String weight = (tank.isSuperHeavy() && !(tank instanceof VTOL)) ? "superheavy" : "standard";
+        return String.format("%s_%s_%s.svg", subtype, turret, weight);
+    }
+
+    @Override
+    protected String getRecordSheetTitle() {
+        StringBuilder sb = new StringBuilder();
+        if (tank.isSupportVehicle()) {
+            // Only take the first word; strip "Support Vehicles"
+            sb.append(tank.getWeightClassName().replaceAll(" .*", " "));
+        } else if (tank.isSuperHeavy()) {
+            sb.append("SuperHeavy ");
+        }
+        switch (tank.getMovementMode()) {
+            case NAVAL:
+            case HYDROFOIL:
+            case SUBMARINE:
+                sb.append("Naval ");
+                break;
+            case VTOL:
+                // do nothing
+                break;
+            default:
+                sb.append(tank.getMovementModeAsString()).append(" ");
+                break;
+        }
+        if (tank.isSupportVehicle()) {
+            sb.append("Support ");
+        }
+        if (tank instanceof VTOL) {
+            sb.append("VTOL");
+        } else {
+            sb.append("Vehicle");
+        }
+        sb.append(" Record Sheet");
+        return sb.toString();
+    }
+
+    @Override
+    protected void writeTextFields() {
+        super.writeTextFields();
+        setTextField(MOVEMENT_TYPE, tank.getMovementModeAsString());
+        setTextField(ENGINE_TYPE, tank.getEngine().getEngineName()
+                .replaceAll("\\d+|\\[.*]", "").trim());
```

Comment:
Reading this Regex, is this intended to match any one or more numbers or the literal `[.*]`?

Reply:
One or more numbers or anything in literal brackets. So "210 XL [Tank Engine]" becomes " XL " which then has the leading and trailing space trimmed.

Reply:
Sounds good.